internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,356 @@
+	var qs []*survey.Question
+
+	message := "Insert the Username for authenticating to MongoDB"
+	dbUser := opts.DBUsername
+	if dbUser == "" {
+		dbUser = dbUsername()
+		if dbUser != "" {
+			message = fmt.Sprintf("Insert the Username for authenticating to MongoDB [Press Enter to use '%s']", dbUser)
+		}
+
+		qs = append(qs, newDBUsernameQuestion(dbUser, message, opts.validateUniqueUsername))
+	}
+
+	if opts.DBUserPassword == "" {
+		qs = append(qs, newDBUserPasswordQuestion())
+	}
+
+	if len(opts.IPAddresses) == 0 {
+		message = "Insert the IP entry to add to the Access List"
+		publicIP := net.IPAddress()
+		if publicIP != "" {
+			message = fmt.Sprintf("Insert the IP entry to add to the Access List [Press Enter to use your public IP address '%s']", publicIP)
+		}
+		q := newAccessListQuestion(publicIP, message)
+		qs = append(qs, q)
+	}
+
+	if len(qs) > 0 {
+		if err := survey.Ask(qs, opts); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.IPAddresses) == 0 && opts.IPAddress != "" {
+		opts.IPAddresses = []string{opts.IPAddress}
+	}
+
+	if opts.DBUserPassword == "" {
+		// The user wants to auto-generate the password
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) validateUniqueUsername(val interface{}) error {
+	username, _ := val.(string)
+	dbUser, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), username)
+	if err != nil {
+		if !strings.Contains(err.Error(), fmt.Sprintf("No user with username %s exists.", username)) {
+			return err
+		}
+	}
+
+	if dbUser != nil {
+		return errors.New("a user with this username already exists")
+	}
+
+	return nil
+}
+
+// dbUsername returns the username of the user by running the command 'whoami'
+func dbUsername() string {
+	userStruct, err := user.Current()
+	if err != nil {
+		return ""
+	}
+
+	// dbUsername can only contain ASCII letters, numbers, hyphens and underscores
+	out := strings.TrimSpace(userStruct.Username)
+	var re = regexp.MustCompile("([^A-Za-z0-9_-])")
+	return re.ReplaceAllString(out, "_")
+}
+
+// newClusterName returns an auto-generate Cluster name
+func (opts *Opts) newClusterName() string {
+	cs, _ := opts.store.ProjectClusters(opts.ConfigProjectID(), nil)
+	i := 0
+	if clusters, ok := cs.([]atlas.Cluster); ok {
+		for {
+			clusterName := "QuickstartCluster" + strconv.Itoa(i)
+			if !search.IsClusterFound(clusters, clusterName) {
+				return clusterName
+			}
+			i++
+		}
+	}
+
+	return ""
+}
+
+// mongocli atlas dbuser(s) quickstart [--clusterName clusterName] [--provider provider] [--region regionName] [--projectId projectId] [--username username] [--password password]
+func Builder() *cobra.Command {
+	opts := &Opts{}
+	cmd := &cobra.Command{
+		Use: "quickstart",
+		Example: `mongocli atlas quickstart
+mongocli atlas quickstart --clusterName Test --provider GPC --username dbuserTest --password Test!
```

Comment:
[nit] `mongocli atlas quickstart` is already part of the help, is better if you explain what this does, ex:
```suggestion
		Example: `
  Skip setting cluster name, provider or database username by using the command options
  $ mongocli atlas quickstart --clusterName Test --provider GPC --username dbuserTest
```